ContentMetadata/Data/DataService.cs

```diff
@@ -35,185 +35,216 @@ SELECT 1
+			{
+				connection.Open();
+
+				using (var command = new SqlCommand(statement, connection))
+				{
+					int.TryParse(command.ExecuteScalar().ToString(), out value);
+				}
+			}
+
+			return Convert.ToBoolean(value);
+		}
+
+		public static bool IsInstalled()
+		{
+			const string statement = @"SELECT 1 FROM SYS.OBJECTS WHERE OBJECT_ID = OBJECT_ID(N'[dbo].[custom_MetaData]') AND TYPE IN (N'U')";
+
+			int value;
+			using (var connection = GetSqlConnection())
+			{
+				connection.Open();
+
+				using (var command = new SqlCommand(statement, connection))
+				{
+					var result = command.ExecuteScalar();
+
+					int.TryParse(result?.ToString(), out value);
+				}
+			}
+
+			return Convert.ToBoolean(value);
+		}
+
+		#endregion
+
+		#region IInstallablePlugin Methods
+
+		internal static void Install()
+		{
+			if (!HasRequiredPermissions())
+			{
+				Apis.Get<IEventLog>().Write("Unable to install SQL scripts for plugin", new EventLogEntryWriteOptions { EventType = "Warning", Category = "ContentMetadata", EventId = 3660 });
+				return;
+			}
+
+			using (var connection = GetSqlConnection())
+			{
+				connection.Open();
+				foreach (var statement in GetBatchedStatementsFromSql())
+				{
+					using (var command = new SqlCommand(statement, connection))
+					{
+						command.ExecuteNonQuery();
+					}
+				}
+			}
+		}
+
+		private static IEnumerable<string> GetBatchedStatementsFromSql()
+		{
+			var sqlScript = GetStatementsFromSql();
+			return Regex.Split(sqlScript, @"^\s*GO\s*$", RegexOptions.IgnoreCase | RegexOptions.Multiline).Select(statement => Regex.Replace(statement, @"(?:^SET\s+.*?$|\/\*.*?\*\/|--.*?$)", "\r\n",
+				RegexOptions.IgnoreCase | RegexOptions.Multiline).Trim()).Where(sanitizedStatement => sanitizedStatement.Length > 0);
+		}
+
+		private static string GetStatementsFromSql()
+		{
+			var stream = typeof(DataService).Assembly.GetManifestResourceStream("ContentMetadata.Resources.Sql.Install.sql");
+			if (stream == null || stream.Length == 0)
+				return string.Empty;
+
+			using (stream)
+			{
+				var data = new byte[stream.Length];
+				stream.Read(data, 0, data.Length);
+				var text = Encoding.UTF8.GetString(data);
+				return text[0] > 255 ? text.Substring(1) : text;
+			}
+		}
+
+		#endregion
+
+		public static List<ContentMetadata> List(Guid contentId)
+		{
+			var items = new List<ContentMetadata>();
+
+			using (var connection = GetSqlConnection())
+			{
+				using (var command = CreateSprocCommand("[custom_MetaData_List]", connection))
+				{
+					command.Parameters.Add("@ContentId", SqlDbType.UniqueIdentifier).Value = contentId;
+
+					connection.Open();
+
+					using (SqlDataReader dr = command.ExecuteReader())
+					{
+						while (dr.Read())
+						{
+							items.Add(Populate(dr));
+						}
+						// Done with the reader and the connection
+						dr.Close();
+					}
+
+					connection.Close();
+				}
+			}
+
+			return items;
+		}
+
+		public static List<ContentMetadata> ListContent(string dataKey)
```

Comment:
Update this method to filter by key & value. Returning all entities with a key and filtering in memory is expensive and won't scale.